sdks/Cordova.md

```diff
@@ -0,0 +1,136 @@
+
+## Proposed solution 1
+
+Pure javascript solution for the mobile configuration and orchestration 
+that will allow cooperability between JavaScript and Native.
+Configuration parser and other pure JS code willnot require any native implementation and can be reused for cordova/react native implementations. Individual SDK will may by default be implemented in pure JavaScript or as Cordova/React plugins. 
+
+## Advantages of the solution 1
+
+- Significantly reduced effort to develop and maintain Cordova Support by reusing available components.
+- Aligned with the way cordova works (single plugin to do specific *native* job, integrated with others in JavaScript)
+- Ability to reuse existing cordova plugins that perform native tasks (push notifications, sync, auth)
+- Ability to wrap existing JavaScript solutions and use them without 
+- Ability to select and use only elements that are needed in the app.
+
+See investigation details bellow for more details
+
+## Proposed solution 2
+
+Pure JavaScript implementations (JavaScript SDK) that will we written for the web and abstract from browser support.
+This means that specific browser functionalities that may not be available for React or Cordova will be provided to enable support for this platforms. JS SDK will be working with Web, Cordova and ReactNative - each platform will provide wrappers for specific apis if needed. This aproach will not allow us to reuse native implementations, but it will make Cordova, Web and ReactNative SDK's more consistent. Implementation could detect platform and load specific ReactNative/Cordova/Browser wrappers.
+
+## Advantages of the solution 2
+
+- Ability to collect and link 3 different platfroms as part of the single repository
+- Code still can be reused thanks to keycloak, push, sync js implementations
+- No need for specific plugins for React and Cordova (platform specific wrappers can be provided) with single code base
+- No need to adjust native code for the plugins
+- No challenges related with cross js/native result access.
+
+See investigation details bellow for more details
+
+## Unanswered questions
+
+Questions that needs to be answered before proposal is merged.
+
+### Collecting all plugins into single repository
+
+Since some of the plugins like push are already in separate repository this can be seen as additional effort. 
+It may also complicate distribution as community preffers to have 1 plugin per repository and cordova cmd is optimized.
+I do not have strong opinions about this one but it feels like leaving things separate and having some aggregator repository with example application that links to all plugins and have parser code will be the best way.
+
+## Architecture for solution 1
+
+### Config parsing 
+
+Cordova config plugin not needed due to trivial implementation that can be done in pure JavaScript.
+Networking/Logging should be done on JavaScript side as this is common aproach for cordova.
```

Comment:
I'm playing devil's advocate here, but could this be a security issue, e.g storing logs securely for upload later? The commonly used storage mechanisms in Cordova and React Native are not encrypted AFAIK. Maybe this is a goal for later?

Reply:
Both react and cordova plug into `console.log` by plugin available by default. We could swap override plugin if anything specific will be needed.